src/migrate/index.js

```diff
@@ -196,7 +196,7 @@ export default class Migrator {
     .then(batchNo => {
       return this._waterfallBatch(batchNo, migrations, direction)
     })
-    .then(() => this._freeLock())
+    .then(retVal => this._freeLock().then(() => retVal))
```

Comment:
I was thinking about it, but noticed that I cannot use `.tap` since if it throws an error control won't go to catch block in promise chain. 

Other reason why I don't like to use it is that I prefer nowadays to use node's native promise since es6 has it now without need to import bluebird.